File: Mcma.Api/McmaApiError.cs
using System;

namespace Mcma.Api
{
    public class McmaApiError
    {
        public McmaApiError(int status = 0, string error = null, string path = null)
        {
            Status = status;
            Error = error;
            Path = path;
        }

        public DateTime Timestamp { get; set; } = DateTime.UtcNow;

        public int Status { get; set; }

        public string Error { get; set; }

        public string Path { get; set; }
    }
}

File: base/api/lib/http/mcma-api-error.ts
import { getStatusError } from "./http-statuses";
import { McmaObject, McmaObjectProperties, Utils } from "@mcma/core";

export interface McmaApiErrorProperties extends McmaObjectProperties {
    timestamp?: Date,
    status: number,
    error?: string,
    path: string,
    message?: string,
}

export class McmaApiError extends McmaObject implements McmaApiErrorProperties {
    timestamp: Date;
    status: number;
    error: string;
    path: string;
    message?: string;

    constructor(properties: McmaApiErrorProperties) {
        super("ApiError");

        this.timestamp = Utils.ensureValidDateOrUndefined(properties.timestamp) ?? new Date();
        this.status = properties.status;
        if (!this.error) {
            this.error = getStatusError(properties.status);
        }
        this.path = properties.path;
        this.message = properties.message;

        Utils.checkProperty(this, "status", "number", true);
        Utils.checkProperty(this, "path", "string", true);
        Utils.checkProperty(this, "message", "string", false);
    }
}
